lib/_http_client.js

```diff
@@ -24,6 +24,18 @@ function ClientRequest(options, cb) {
     if (!options.hostname) {
       throw new Error('Unable to determine the domain name');
     }
+  } else if (options instanceof url.URL) {
+    options = {
+      protocol: options.protocol,
+      host: options.host,
+      hostname: options.hostname,
+      port: options.port,
+      hash: options.hash,
+      search: options.search,
+      pathname: options.pathname,
+      path: `${options.pathname}${options.search}`,
+      href: options.href
+    };
```

Comment:
I can eliminate the duplication but having the instanceof and copying into a separate options is the most reliable and requires the fewest number of changes throughout the code.